### PR TITLE
Add MindsDB dbt adapter

### PR DESCRIPTION
**Pull request by ZoranPandovski**
## Description & motivation

Add MindsDB adapter.  [MindsDB](https://mindsdb.com/) is an open-source framework that enables machine learning workflows for the most powerful databases and data warehouses using SQL. Related to https://github.com/mindsdb/dbt-mindsdb/issues/1

## Checklist
If you added new pages (delete if not applicable):
- [X] The page has been added to `website/sidebars.js`
- [X] The new page has a unique filename

### CODE REVIEW

File: website/docs/reference/warehouse-profiles/mindsdb-profile.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+| Key      | Required | Description                                          | Example                        |
+| -------- | -------- | ---------------------------------------------------- | ------------------------------ |
+| type     |    ✔️   | The specific adapter to use                          | `mindsdb`                      |
+| host     |    ✔️   | The MindsDB (hostname) to connect to                 | `cloud.mindsdb.com`            |
+| port     |    ✔️   | The port to use                                      | `3306`  or `47335`             |
+| schema   |    ✔️   | Specify the schema (database) to build models into   | The MindsDB datasource         |
+| username |    ✔️   | The username to use to connect to the server         | `mindsdb` or mindsdb cloud user|
+| password |    ✔️   | The password to use for authenticating to the server | `pass                          |
+
+## Usage
```
Comment (dataders) [inline website/docs/reference/warehouse-profiles/mindsdb-profile.md]: can you add this information to a separate config page as described [here](https://docs.getdbt.com/docs/contributing/documenting-a-new-adapter#topics-and-pages-to-cover)?

File: website/docs/reference/warehouse-profiles/mindsdb-profile.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+## Testing
+
+- Install dev requirements
+```    
+  pip install -r dev_requirements.txt
+```
+- Run pytest
+```    
+  python -m pytest tests/
+```
```
Comment (dataders) [inline website/docs/reference/warehouse-profiles/mindsdb-profile.md]: not sure this section is relevant for end-users, probably makes sense to keep in the GitHub repo's `README.md` or, better yet, a `CONTRIBUTING.md`.
```suggestion
## Testing

- Install dev requirements
```    
  pip install -r dev_requirements.txt
```
- Run pytest
```    
  python -m pytest tests/
```
```

File: website/docs/reference/warehouse-profiles/mindsdb-profile.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+If you need to specify schema you can do it with dot separator: schema_name.table_name.sql    
+Parameters:
+  - predictor_name - name of using predictor.
+    It has to be created in mindsdb
+  - integration - name of used integration to get data from and save result to.
+    In has to be created in mindsdb beforehand
+```    
+    {{ config(materialized='table', predictor_name='TEST_PREDICTOR_NAME', integration='int1') }}
+        select a, bc from ddd where name > latest
+```
```
Comment (dataders) [inline website/docs/reference/warehouse-profiles/mindsdb-profile.md]:
```suggestion
Create dbt project, choose mindsdb as the database and set up the connection. Verify your connection works `dbt debug`

`dbt init <project_name>`

To create a predictor, create a dbt model with a "predictor" materialization. The name of the model will be the name of predictor.

#### Parameters:
- `integration` - name of used integration to get data from and save result to. Must be created in mindsdb beforehand
- `predict` - field for prediction
- `predict_alias` [optional] - alias for predicted field
- `using` [optional] - options for configure trained model

```sql
-- my_first_model.sql    
    {{
        config(
            materialized='predictor',
            integration='photorep',
            predict='name',
            predict_alias='name',
            using={
                'encoders.location.module': 'CategoricalAutoEncoder',
                'encoders.rental_price.module': 'NumericEncoder'
            }
        )
    }}
      select * from stores
```

To apply predictor add dbt model with "table" materialization. It creates or replaces table in selected integration with results of predictor. The models name will be the name of the resulting predictions table. If you need to specify schema you can do it with dot separator: `schema_name.table_name.sql`
    
#### Parameters:
- `predictor_name` - name of using predictor.
    It has to be created in mindsdb
- `integration` - name of used integration to get data from and save result to (must be created in mindsdb beforehand)

```sql
-- my_models_predictions.sql 
    {{ config(materialized='table', predictor_name='TEST_PREDICTOR_NAME', integration='int1') }}
        select a, bc from ddd where name > latest
```
```

File: website/docs/reference/warehouse-profiles/mindsdb-profile.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+- Create dbt project, choose mindsdb database and set up connection
+```    
+    dbt init <project_name>
+```
+- To create predictor add dbt model with "predictor" materialization: 
+Name of the model is used as name of predictor.
+Parameters:
+  - integration - name of used integration to get data from and save result to.
+    In has to be created in mindsdb beforehand
```
Comment (dataders) [inline website/docs/reference/warehouse-profiles/mindsdb-profile.md]: how do you create the integration? can you at least link to a page that shows how to do it?

File: website/docs/reference/warehouse-profiles/mindsdb-profile.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            }
+        )
+    }}
+      select * from stores
+```
+
+- To apply predictor add dbt model with "table" materialization. 
+It creates or replaces table in selected integration with results of predictor.
+Name of the model is used as name of the table to store prediction results. 
+If you need to specify schema you can do it with dot separator: schema_name.table_name.sql
```
Comment (dataders) [inline website/docs/reference/warehouse-profiles/mindsdb-profile.md]: can the user not also configure the schema by using a `schema` parameter in `config`? If yes, then please say so. If not, then I'd like to know why!

File: website/docs/reference/warehouse-profiles/mindsdb-profile.md (hunk 3 above)
Comment (dataders) [reply website/docs/reference/warehouse-profiles/mindsdb-profile.md]: this suggestion above got a little kooky, but the major changes I made (beyond edits for clarity) were:
- no nested bullets, only bullets for the parameters
- adding `sql` to the beginning of of codeblocks
- adding a comment with an example file name for the sql

**Review by dataders — CHANGES_REQUESTED** [review]
this is a great first draft! I've got some small changes to propose, but this is almost ready!

File: website/docs/reference/warehouse-profiles/mindsdb-profile.md (hunk 5 above)
Comment (ZoranPandovski) [reply website/docs/reference/warehouse-profiles/mindsdb-profile.md]: mindsdb can have several integrations and several schemas (in one dbt project). If we choose to set up the schema in the config we can have only one schema.

File: website/docs/docs/available-adapters.md [hunk 6]
```diff
@@ -38,6 +38,7 @@ These adapter plugins are built and maintained by the same people who build and
 | Databricks ([dbt-databricks](https://github.com/databricks/dbt-databricks)) | [Profile Setup](databricks-profile), [Configuration](spark-configs#databricks-configurations) | `pip install dbt-databricks` |
 | Firebolt ([dbt-firebolt](https://github.com/firebolt-db/dbt-firebolt)) | [Profile Setup](firebolt-profile), [Configuration](firebolt-configs) | `pip install dbt-firebolt` |
 | Materialize ([dbt-materialize](https://github.com/MaterializeInc/materialize/blob/main/misc/dbt-materialize))  | [Profile Setup](materialize-profile), [Configuration](materialize-configs) | `pip install dbt-materialize` |
+| MindsDB ([dbt-mindsdb](https://github.com/mindsdb/dbt-mindsdb))  | [Profile Setup](mindsdb-profile)       | `pip install dbt-mindsdb`     |
```
Comment (dataders) [inline website/docs/docs/available-adapters.md]:
```suggestion
| MindsDB ([dbt-mindsdb](https://github.com/mindsdb/dbt-mindsdb))  | [Profile Setup](mindsdb-profile), [Configuration](mindsdb-configs)      | `pip install dbt-mindsdb`     |
```

File: website/docs/reference/warehouse-profiles/mindsdb-profile.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+If you need to specify schema you can do it with dot separator: schema_name.table_name.sql  
+
+#### Parameters
+- `predictor_name` - name of the predictor. It has to be created in mindsdb.
+- `integration` - name of used integration to get data from and save result to. Must be created in mindsdb beforehand using the [`CREATE DATABASE` syntax](https://docs.mindsdb.com/sql/create/databases/).
+
+```    
+    {{ config(materialized='table', predictor_name='TEST_PREDICTOR_NAME', integration='photorep') }}
+        select a, bc from ddd where name > latest
+```
```
Comment (dataders) [inline website/docs/reference/warehouse-profiles/mindsdb-profile.md]: can you move this whole Usage section to `mindsdb-configs.md`? configs is how to use the adapter. profile is how to connect.

**Review by dataders — CHANGES_REQUESTED** [review]
almost there!

**Review by dataders — APPROVED** [review]
beautiful -- thanks @ZoranPandovski!

**Pull request merged**